icu4j/main/classes/core/src/com/ibm/icu/impl/Utility.java

```diff
@@ -1858,4 +1859,32 @@ public static int charSequenceHashCode(CharSequence value) {
             throw new ICUUncheckedIOException(e);
         }
     }
+
+    /**
+     * Java 8+ String#join(CharSequence, Iterable<? extends CharSequence>) compatible method for Java 7 env.
+     * @param delimitter the delimiter that separates each element
```

Comment:
Please fix these typos in multiple places: "delimitter", "Delimitter" (only one 't' each)